Description: Fix testsuite
Author: Bastien ROUCARIÈS <roucaries.bastien@gmail.com>
Forwarded: no
Last-Update: 2025-09-14

--- a/test/vectors.js
+++ b/test/vectors.js
@@ -11,6 +11,15 @@
 	var v = vectors[i];
 
 	tape(alg + ': NIST vector ' + i, function (t) {
+		// nan test
+		if(v!=v) {
+			t.end();
+			return;
+		}
+		if(typeof(v)=='undefined') {
+			t.end();
+			return;
+		}
 		if (verbose) {
 			t.comment(v);
 			t.comment('VECTOR', i);
